control-plane/pkg/apis/sources/v1beta1/kafka_types.go

```diff
@@ -82,6 +83,12 @@ type KafkaSourceSpec struct {
 	// +optional
 	Delivery *eventingduckv1.DeliverySpec `json:"delivery,omitempty"`
 
+	// Ordering is the type of the consumer verticle.
+	// Should be ordered or unordered.
+	// By default, it is ordered.
+	// +optional
+	Ordering eventing.DeliveryOrdering `json:"ordering,omitempty"`
```

Comment:
We should:
- add validation for this field in `Validate`
- add defaulting for this field in `SetDefaults`
- since it's optional I'd make it a pointer instead of relying on "empty string == unset"